routing/turns_tts_text.cpp

```diff
@@ -45,10 +46,195 @@ void GetTtsText::ForTestingSetLocaleWithJson(std::string const & jsonBuffer, std
   m_getCurLang = platform::ForTestingGetTextByIdFactory(jsonBuffer, locale);
 }
 
+// Returns -1 for not found, positive for which item in the haystack was found
+long FindInStrArray(const std::vector<std::string>& haystack, std::string& needle)
+{
+  auto begin = haystack.begin();
+  for(std::vector<std::string>::const_iterator i = begin; i != haystack.end(); ++i) {
+    std::string item = *i;
+    std::string::size_type idx = needle.find(item);
+    if (idx != std::string::npos) {
+      long item_i = std::distance( begin, i );
+      LOG(LINFO, ("TTSf found", item_i, item, idx, needle));
+      return item_i;
+    }
+  }
+  return -1;
+}
+
+void HungarianBaseWordTransform(std::string& myString) {
+  std::string_view baseStr[] = {"e", "a", "ö", "ü"};
+  std::string_view harmonyStr[] = {"é", "á", "ő", "ű"};
+
+  for (size_t i = 0; i < std::size(baseStr); ++i)
+  {
+    if (strings::EndsWith(myString, baseStr[i]))
+      myString.replace(myString.size() - baseStr[i].size(), baseStr[i].size(), harmonyStr[i]);
+  }
+}
+
+uint8_t CategorizeHungarianAcronymsAndNumbers(std::string const & myString)
+{
+  std::vector<std::string> backNames = {
```

Comment:
The memory at runtime is allocated 1) for the vector and 2) for each constant string each time this function is run. Please treat OM as an embedded environment where each overhead matters.

Use constexpr std::array and std::string_view instead.

Or even better, if these values are used for the search, please create a regex instead. It will work faster.